src/index.js

```diff
@@ -21,8 +23,23 @@ export default class extends Controller {
       addSuffix: this.data.has('addSuffix') || false
     }
 
+    let formattedDate
```

Comment:
That's a good idea!

What about this simpler version:

```js
  load () {
    const datetime = this.data.get('datetime')
    const date = Date.parse(datetime)
    const options = {
      includeSeconds: this.data.has('includeSeconds') || false,
      addSuffix: this.data.has('addSuffix') || false
    }

    if (Number.isNaN(date)) {
      this.isValid = false

      console.error(
        `Value given in 'data-timeago-datetime' is not a valid date (${datetime}). Please provide a ISO 8601 compatible datetime string. Displaying given value instead.`
      )
    }

    this.element.dateTime = datetime
    this.element.innerHTML = this.isValid ? formatDistanceToNow(date, options) : datetime
  }
```

Reply:
Hmm. I think this won't work, because `Date.parse(datetime)` will throw the error and exit there. So all after that isn't executed. That's why we need the `try catch` here

Reply:
Something like this would work:
```javascript
  load () {
    const datetime = this.data.get('datetime')
    const options = {
      includeSeconds: this.data.has('includeSeconds') || false,
      addSuffix: this.data.has('addSuffix') || false
    }

    let date
    try {
      date = Date.parse(datetime)
    } catch (e) {
      this.isValid = false
      console.error(
        `Value given in "data-timeago-datetime" is not a valid date (${datetime}). Please provide a ISO 8601 compatible datetime string. Displaying given value instead.`
      )
    }

    this.element.dateTime = datetime
    this.element.innerHTML = this.isValid ? formatDistanceToNow(date, options) : datetime
  }
```

Reply:
`formatDistanceToNow` throw an exception if the date is invalid not the `parse` actually.

I found in the [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) that `parse` returns `NaN` for invalid date.

This code works 👍

Reply:
Sorry, you are right! I mixed up `Date.parse()` with `new Date()`. Will update the PR!